File: split/printspec.pas
procedure printspec(p: integer; s: strnumber);
begin
  if (p < memmin) or (p >= lomemmax) then
    printchar(42)
  else
  begin
    printscaled(mem[p + 1].int);
    if s <> 0 then
      print(s);
    if mem[p + 2].int <> 0 then
    begin
      print(312);
      printglue(mem[p + 2].int, mem[p].hh.b0, s);
    end;
    if mem[p + 3].int <> 0 then
    begin
      print(313);
      printglue(mem[p + 3].int, mem[p].hh.b1, s);
    end;
  end;
end;

File: cpp/printspec.cpp
#include "printspec.h"
#include "printchar.h"
#include "print.h"
#include "printglue.h"
#include "printscaled.h"
#include "texte.h"

void printspec(int p, const std::string &s)
{
	if (p < memmin || p >= lomemmax)
		printchar('*');
	else
	{
		printscaled(width(p));
		if (txt(s))
			print(s);
		if (stretch(p))
		{
			print(" plus ");
			printglue(stretch(p), type(p), s);
		}
		if (shrink(p))
		{
			print(" minus ");
			printglue(shrink(p), subtype(p), s);
		}
	}
}
